api4/group.go

```diff
@@ -505,6 +509,37 @@ func getGroupMembers(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Write(b)
 }
 
+func getGroupsByUserId(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireUserId()
+	if c.Err != nil {
+		return
+	}
+
+	if c.App.Session().UserId != c.Params.UserId && !c.App.SessionHasPermissionTo(*c.App.Session(), model.PERMISSION_MANAGE_SYSTEM) {
```

Comment:
Shouldn't this be a `||`? We would want to deny permission if the user requesting the groups is either not the user with the UserID in the params OR is not a system admin? So only grant access if we are a system admin or the user with the userID in the params.

Reply:
No, && is correct. 
If the contextual user is NOT the parameter user AND NOT a System Admin. - Deny Access.
I have unit tests that check all three scenarios.
Its actually the same permission check in `getTeamsForUser()`

Reply:
@sbishel Sorry, you are right. I made a mistake and didn't think carefully enough about it. Thanks for the correction.